pilot/pkg/security/authz/builder/builder_test.go

```diff
@@ -0,0 +1,237 @@
+		Namespaces: []string{"special"},
+	}
+	cfg1 := newAuthzPolicyWithRbacConfig(istio_rbac.RbacConfig_ON, nil, nil)
+	cfg2 := newAuthzPolicyWithRbacConfig(istio_rbac.RbacConfig_OFF, nil, nil)
+	cfg3 := newAuthzPolicyWithRbacConfig(istio_rbac.RbacConfig_ON_WITH_INCLUSION, target, nil)
+	cfg4 := newAuthzPolicyWithRbacConfig(istio_rbac.RbacConfig_ON_WITH_EXCLUSION, nil, target)
+	cfg5 := newAuthzPolicyWithRbacConfig(istio_rbac.RbacConfig_ON, nil, nil)
+
+	testCases := []struct {
+		Name          string
+		AuthzPolicies *model.AuthorizationPolicies
+		Service       string
+		Namespace     string
+		want          bool
+	}{
+		{
+			Name:          "rbac plugin enabled",
+			AuthzPolicies: cfg1,
+			want:          true,
+		},
+		{
+			Name:          "rbac plugin disabled",
+			AuthzPolicies: cfg2,
+		},
+		{
+			Name:          "rbac plugin enabled by inclusion.service",
+			AuthzPolicies: cfg3,
+			Service:       "product.default.svc",
+			Namespace:     "default",
+			want:          true,
+		},
+		{
+			Name:          "rbac plugin enabled by inclusion.namespace",
+			AuthzPolicies: cfg3,
+			Service:       "other.special.svc",
+			Namespace:     "special",
+			want:          true,
+		},
+		{
+			Name:          "rbac plugin disabled by exclusion.service",
+			AuthzPolicies: cfg4,
+			Service:       "product.default.svc",
+			Namespace:     "default",
+		},
+		{
+			Name:          "rbac plugin disabled by exclusion.namespace",
+			AuthzPolicies: cfg4,
+			Service:       "other.special.svc",
+			Namespace:     "special",
+		},
+		{
+			Name:          "rbac plugin enabled with permissive",
+			AuthzPolicies: cfg5,
+			want:          true,
+		},
+	}
+
+	for _, tc := range testCases {
+		got := isRbacEnabled(tc.Service, tc.Namespace, tc.AuthzPolicies)
+		if tc.want != got {
+			t.Errorf("%s: expecting %v but got %v", tc.Name, tc.want, got)
+		}
+	}
+}
+
+func TestBuilder_BuildHTTPFilter(t *testing.T) {
+	service := newService("bar.a.svc.cluster.local", nil, t)
+
+	testCases := []struct {
+		name                        string
+		isXDSMarshalingToAnyEnabled bool
+		policies                    []*model.Config
+		wantRuleWithPolicies        bool
+	}{
+		{
+			name:                        "XDSMarshalingToAnyEnabled",
+			isXDSMarshalingToAnyEnabled: true,
+		},
+		{
+			name: "HTTP rule",
+			policies: []*model.Config{
+				simpleRole("role-1", "a", "bar"),
+				simpleBinding("binding-1", "a", "role-1"),
+			},
+			wantRuleWithPolicies: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		policy := newAuthzPolicies(tc.policies, t)
+		b := NewBuilder(service, policy, tc.isXDSMarshalingToAnyEnabled)
+
+		got := b.BuildHTTPFilter()
+		if got.Name != authz_model.RBACHTTPFilterName {
+			t.Errorf("%s: got filter name %q but want %q", tc.name, got.Name, authz_model.RBACHTTPFilterName)
+		}
+
+		if tc.isXDSMarshalingToAnyEnabled {
+			if got.GetTypedConfig() == nil {
+				t.Fatalf("%s: want typed config when isXDSMarshalingToAnyEnabled is true", tc.name)
```

Comment:
Why do we want to exit the test if one test case fails?

Reply:
yeah, it's better to change this to `t.Errorf()`,